discovery_engine/lib/src/infrastructure/migration.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:xayn_discovery_engine/src/domain/models/active_search.dart';
+import 'package:xayn_discovery_engine/src/domain/models/document.dart';
+import 'package:xayn_discovery_engine/src/domain/models/source.dart';
+import 'package:xayn_discovery_engine/src/domain/models/source_reacted.dart';
+import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_active_document_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_active_search_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_document_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_engine_state_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_source_preference_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_source_reacted_repo.dart';
+
+class DartMigrationData {
+  final Uint8List? engineState;
+  final List<Document> documents;
+  final Map<DocumentId, ActiveDocumentData> activeDocumentData;
+  final List<SourceReacted> reactedSources;
+  final Set<Source> trustedSources;
+  final Set<Source> excludedSources;
+  final ActiveSearch? activeSearch;
+
+  /// Cleanup callback (which should be) called after successfully initializing the engine.
+  final Future<void> Function() cleanup;
+
+  DartMigrationData({
+    required this.engineState,
+    required this.documents,
+    required this.activeDocumentData,
+    required this.reactedSources,
+    required this.trustedSources,
+    required this.excludedSources,
+    required this.activeSearch,
+    required this.cleanup,
+  });
+
+  /// If migrations is necessary extracts the data for it from the repos.
+  ///
+  /// The `cleanup` callback is set to clear the repositories.
+  static Future<DartMigrationData?> fromRepositories(
+    HiveEngineStateRepository engineStateRepository,
+    HiveDocumentRepository documentRepository,
+    HiveActiveSearchRepository activeSearchRepository,
+    HiveActiveDocumentDataRepository activeDocumentDataRepository,
+    HiveSourceReactedRepository sourceReactedRepository,
+    HiveSourcePreferenceRepository sourcePreferenceRepository,
+  ) async {
+    if (engineStateRepository.box.isEmpty &&
+        documentRepository.box.isEmpty &&
+        activeSearchRepository.box.isEmpty &&
+        activeDocumentDataRepository.box.isEmpty &&
+        //FIXME even with cfgFeatureStorage we still set this values to hive
+        // sourcePreferenceRepository.box.isEmpty &&
+        sourceReactedRepository.box.isEmpty) {
+      return null;
+    }
+
+    return DartMigrationData(
+      engineState: await engineStateRepository.load(),
+      documents: await documentRepository.fetchAll(),
+      activeDocumentData: activeDocumentDataRepository.box.toMap().map(
+            // ignore: avoid_annotating_with_dynamic
+            (dynamic key, data) =>
```

Comment:
Maybe you can try without dynamic
```suggestion
            (Object key, data) =>
```

Reply:
sadly that doesn't work :sweat:

Reply:
maybe try `Object?` but if that doesn't work either, then just leave it as is